extras/src/main/java/com/novoda/espresso/AccessibilityServiceToggler.java

```diff
@@ -0,0 +1,87 @@
+package com.novoda.espresso;
+
+import android.content.ContentResolver;
+import android.provider.Settings;
+
+class AccessibilityServiceToggler {
+
+    private static final String EMPTY_STRING = "";
+    private static final String SERVICES_SEPARATOR = ":";
+
+    private final SecureSettings secureSettings;
+
+    static AccessibilityServiceToggler create(ContentResolver contentResolver) {
+        return new AccessibilityServiceToggler(new SecureSettings(contentResolver));
+    }
+
+    AccessibilityServiceToggler(SecureSettings secureSettings) {
+        this.secureSettings = secureSettings;
+    }
+
+    void enable(Service service) {
+        String enabledServices = secureSettings.enabledAccessibilityServices();
+        if (enabledServices.contains(service.qualifiedName)) {
+            return;
+        }
+        secureSettings.enabledAccessibilityServices(enabledServices + SERVICES_SEPARATOR + service.qualifiedName);
+    }
+
+    void disable(Service service) {
+        String enabledServices = secureSettings.enabledAccessibilityServices();
+        if (!enabledServices.contains(service.qualifiedName)) {
+            return;
+        }
+
+        String remainingServices = enabledServices
```

Comment:
considered using regex for this but would still need to construct the pattern:

```java
String.replaceAll("(:" + service.qualifiedName + "|" + service.qualifiedName + ":|" + service.qualifiedName + ")", EMPTY_STRING)
```

so this seems clearer.

Reply:
This is covered by tests